Remove use of "origin" for non-HTTP schemes (#204)
MikeBishop/dns-alt-svc: draft-ietf-dnsop-svcb-https.md

This document specifies the "SVCB" and "HTTPS" DNS resource record (RR)
types to facilitate the lookup of information needed to make connections

> to network services, such as for HTTPS URLs.  SVCB records

Reviewer: ```suggestion
to network services, such as for HTTPS origins.  SVCB records
```
Not text changed by this PR, but I'd probably talk about the HTTPS origin as one of the services being described, rather than the URL.  In fact, this is explicitly stated further down.

The goal of the SVCB RR is to allow clients to resolve a single
additional DNS RR in a way that:

> * Provides alternative endpoints that are authoritative for the service,

Reviewer: While this doesn't need to be addressed before merging, I'm not certain that the concept of "authoritative" is the same in all possible protocols, particularly ones where the scope of a "service" might vary.  We might want to consider whether that's a term we want to keep as well.

## Terminology

> 

Reviewer 2: Add "service name"?  (The hostname within the authority.)

For consistency with {{AltSvc}}, we adopt the following definitions:
Our terminology is based on the common case where the SVCB record is used to
access a resource identified by a URI whose `authority` field contains a DNS
hostname as the `host`.

* An "origin" is an information source as in {{!RFC6454}}.
  For services other than HTTPS, the exact definition will
  need to be provided by the document mapping that service
  onto the SVCB RR.
* The "origin server" is the server that the client would reach when
  accessing the origin in the absence of the SVCB record
  or an HTTPS Alt-Svc.

> * An "alternative service" is a different server that can serve the

Reviewer 2: It sounds like "authority endpoint"?

  origin over a specified protocol.

> For example within HTTPS, the origin consists of a scheme (typically

Reviewer 2: Do we want to keep this illustrative example but reword it?  I think having an example here would help.

> * The "authority address" is the authority's hostname and a port number implied

Reviewer 2: Given how we use this, either "default endpoint" or "implicit endpoint" might be better.

> * The "authority address" is the authority's hostname and a port number implied

Author: How about "authority endpoint"?

> * The "authority address" is the authority's hostname and a port number implied

Reviewer 2: That's better.

> * An "alternative endpoint" is a hostname, port number, and other associated
>   instructions to the client on how to reach the service.

Reviewer 2: ```suggestion
* An "alternative endpoint" is a hostname, port number, and other associated
  instructions to the client on how to reach an instance of service.
```

# Client behavior {#client-behavior}

> An SVCB-aware client resolves a service with hostname $HOST by attempting

Reviewer 2: ```suggestion
A SVCB-aware client resolves a service with hostname $HOST by attempting
```

> to determine
> the preferred SvcFieldValue and IP addresses for its service, using the
> following procedure:

Reviewer 2: ```suggestion
to determine endpoint (the preferred SvcFieldValue and IP addresses for its service),
using the following procedure:
```

> 1. Let $SVCB_QNAME be $HOST, plus appropriate prefixes for the scheme
>    (see {{svcb-names}}).

Reviewer 2: ```suggestion
1. Let $SVCB_QNAME be $HOST, plus appropriate prefixes for the scheme
   (see {{svcb-names}}).  Let $HOST_QNAME be $HOST.
```

Separate out $SVCB_HOST so we don't mutate $HOST in step 3?

> 2. Issue parallel AAAA/A queries for $HOST and a SVCB query for $SVCB_QNAME.

Reviewer 2: ```suggestion
2. Issue parallel AAAA/A queries for $HOST_QNAME and a SVCB query for $SVCB_QNAME.
```

   The answers for these may or may not include CNAME pointers
   before reaching one or more of these records.

3. If a SVCB record of AliasForm SvcRecordType is returned for $SVCB_QNAME,

>    clients MUST set $HOST and $SVCB_QNAME to SvcDomainName (without

Reviewer 2: ```suggestion
   clients MUST set $HOST_QNAME and $SVCB_QNAME to SvcDomainName (without
```

   additional prefixes) and loop back to step 2,
   subject to chain length limits and loop detection heuristics (see
   {{client-failures}}).

4. If one or more SVCB records of ServiceForm SvcRecordType are returned
   for $SVCB_QNAME, clients should select the highest-priority option with
   acceptable parameters, and resolve AAAA and/or A records for its
   SvcDomainName if they are not already available.  These are the
   preferred SvcFieldValue and IP addresses.  If the connection fails, the
   client MAY try to connect using values from a lower-priority record.
   If none of the options succeed, the client SHOULD connect to the authority

>    address directly.

Reviewer 2: Should it connect to the authority address or $HOST_QNAME?  The latter implies that the AliasForm SVCB record should be ignored if ServiceForm SVCB records were present and didn't work.

Reviewer 2: (This is an interesting question which may want to be its own issue for discussion.)

>    If none of the options succeed, the client SHOULD connect to the authority
>    address directly.

Reviewer 2: ```suggestion
   If none of the options succeed, the client SHOULD connect to $HOST_QNAME directly.
   If that fails, the client SHOULD fall back to $HOST.
```
(SHOULD or MAY for the final fall back to $HOST?)

> 5. If a SVCB record for $SVCB_QNAME does not exist, the received AAAA and/or A
>    records are the preferred IP addresses and there is no SvcFieldValue.

Reviewer 2: ```suggestion
5. If a SVCB record for $SVCB_QNAME does not exist, the received AAAA and/or A
   records from $HOST_QNAME are the preferred IP addresses 
   and there is no SvcFieldValue.
```
